libraries/cmd/unit_generic.h

```diff
@@ -990,13 +991,23 @@ class Unit : public Armed, public Audible, public Drawable, public Damageable, p
         return filename.get();
     }
 
-//Is this class a unit
+private:
+    void setUnitKey(const std::string& key) {
+        unit_key = key;
+        // Cache this calculation so that we don't have to compare the strings over and over again
+        is_planet = boost::algorithm::ends_with(unit_key, "__planets");
```

Comment:
This is much more fragile. Better to use the constructor.
Of course, my suggestion only works if we use `Planet` class for planets.

Reply:
Yeah. I'm not sure that we do use the `Planet` class for planets.

I verified that all of the places in the code that set unit_key use this method now. So `is_planet` will always be set correctly. (Or its replacement, `unit_type`, as you noted above.)

Reply:
We have only two choices with our implementation: either it can be fragile, or it can be unnecessarily computation-heavy. Seems to me

Reply:
> This is much more fragile. Better to use the constructor. Of course, my suggestion only works if we use `Planet` class for planets.

if we don't, we probably should.

Reply:
> > This is much more fragile. Better to use the constructor. Of course, my suggestion only works if we use `Planet` class for planets.
> 
> if we don't, we probably should.

That's a bigger change that I'm not prepared to make in this PR.